test/unit/mdc-top-app-bar/mdc-top-app-bar.test.js

```diff
@@ -0,0 +1,114 @@
+import td from 'testdouble';
+
+import {MDCTopAppBar} from '../../../packages/mdc-top-app-bar';
+
+function getFixture() {
+  return bel`
+    <div>
+        <header class="mdc-top-app-bar">
+      <div class="mdc-top-app-bar__row">
+        <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start">
+          <a href="#" class="material-icons mdc-top-app-bar__menu-icon">menu</a>
+          <span class="mdc-top-app-bar__title">Title</span>
+        </section>
+        <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end" 
+        role="top-app-bar">
+          <a href="#" class="material-icons mdc-top-app-bar__icon" aria-label="Download" alt="Download">
+          file_download</a>
+          <a href="#" class="material-icons mdc-top-app-bar__icon" aria-label="Print this page" alt="Print this page">
+          print</a>
+          <a href="#" class="material-icons mdc-top-app-bar__icon" aria-label="Bookmark this page" 
+          alt="Bookmark this page">bookmark</a>
+          <div class="mdc-menu-anchor">
+            <div class="mdc-menu" tabindex="-1" id="demo-menu">
+              <ul class="mdc-menu__items mdc-list" role="menu" aria-hidden="true" style="transform: scale(1, 1);">
+              </ul>
+            </div>
+          </div>
+        </section>
+      </div>
+    </header>
+      <main class="mdc-top-app-bar-fixed-adjust">
+      </main>
+    </div>
+  `;
+}
+
+function setupTest() {
+  const fixture = getFixture();
+  const root = fixture.querySelector('.mdc-top-app-bar');
+  const adjust = fixture.querySelector('.mdc-top-app-bar-fixed-adjust');
+  const component = new MDCTopAppBar(root);
+  return {root, adjust, component};
+}
+
+suite('MDCTopAppBar');
+
+test('attachTo initializes and returns an MDCTopAppBar instance', () => {
+  assert.isTrue(MDCTopAppBar.attachTo(getFixture()) instanceof MDCTopAppBar);
+});
+
+test('adapter#hasClass returns true if the root element has specified class', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+  assert.isTrue(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter#hasClass returns false if the root element does not have specified class', () => {
+  const {component} = setupTest();
+  assert.isFalse(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {root, component} = setupTest();
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isTrue(root.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isFalse(root.classList.contains('foo'));
+});
+
+test('#adapter.registerNavigationIconInteractionHandler adds a handler to the nav icon ' +
```

Comment:
How hard would it be to add tests to verify that no handler gets registered if there is no navigation icon? Currently the implicit "else" branch isn't covered in the register/deregister logic.